runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -384,11 +399,38 @@ class ROMClassCreationContext
 	{
 		if (!isRedefining() && !isRetransforming()) {
 			if (NULL != _className) {
+				U_16 classNameLenToCompare0 = (U_16)_classNameLength;
+				U_16 classNameLenToCompare1 = classNameLength;
+				BOOLEAN misMatch = FALSE;
 				if (isClassHidden()) {
-					/* for hidden class className has ROM address appended at the end, _className does not have that */
-					classNameLength = (U_16)_classNameLength;
+					if (isROMClassShareable()) {
+						U_8* lambdaClass0 = (U_8*)getLastDollarSignOfLambdaClassName((const char*)_className, _classNameLength);
+						U_8* lambdaClass1 = (U_8*)getLastDollarSignOfLambdaClassName((const char*)className, classNameLength);
+						if ((NULL != lambdaClass0) 
+							&& (NULL != lambdaClass1)
+						) {
+							/**
+							 * Lambda class has class name: HostClassName$$Lambda$<IndexNumber>/0x0000000000000000. 
+							 * Do not need to compare the IndexNumber as it can be different from run to run.
+							 */
+							classNameLenToCompare0 = (U_16)(lambdaClass0 - _className + 1);
+							classNameLenToCompare1 = (U_16)(lambdaClass1 - className + 1);
+						} else if ((NULL == lambdaClass0)
+								&& (NULL == lambdaClass1)
```

Comment:
I see no good reason to use a multi-line if here.

Reply:
Move them into a single line.